core/unit_test/TestReducers.hpp

```diff
@@ -319,6 +329,89 @@ struct TestReducers {
       value = value || values(i);
     }
   };
+
+  // get number of teams for TeamPolicy depending on the tested type
+  constexpr static int get_num_teams() {
+    if constexpr (sizeof(Scalar) == 1) {
+      return 126;
+    } else if constexpr (std::is_same_v<Scalar,
+                                        Kokkos::Experimental::bhalf_t>) {
+      return 256;
+    }
+
+    return 1024;
+  }
+
+  static void test_sum_team_policy(int N, SumFunctor f, Scalar reference_sum) {
+#ifdef KOKKOS_ENABLE_OPENACC
+    if constexpr (std::is_same_v<ExecSpace, Kokkos::Experimental::OpenACC> &&
+                  (std::is_same_v<Scalar, size_t> ||
+                   std::is_same_v<Scalar, double>)) {
+      return;  // FIXME_OPENACC
+    }
+#endif
+#ifdef KOKKOS_ENABLE_OPENMPTARGET
+    if constexpr (std::is_same_v<ExecSpace,
+                                 Kokkos::Experimental::OpenMPTarget>) {
+      return;  // FIXME_OPENMPTARGET
```

Comment:
TODO: narrow this down to skip the tests for specific types only (if there aren't too many of them).